components/optional/index.ts

```diff
@@ -1,7 +1,15 @@
 import PropTypes from 'prop-types';
```

Comment:
I actually think that as part of converting something to TypeScript it makes sense to get rid of all the PropTypes stuff.